File: LeetCode Problems/301 to 400/392. Is Subsequence/392. Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int pos = 0;
        for (int i = 0; i < t.size(); i++) {
            if (t[i] == s[pos])
                pos++;
        }
        if (pos == s.size())
            return true;
        return false;
    }
};

File: LeetCode Problems/301 to 400/392. Is Subsequence/392. Is Subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int j = 0;
        for (int i = 0; i < t.length() && j < s.length(); i++) {
            if (t.charAt(i) == s.charAt(j)) j++;
        }
        if (j == s.length()) return true;
        return false;
    }
}
